src/pymatgen/io/lobster/lobsterenv.py

```diff
@@ -955,50 +953,71 @@ def _find_environments(
                     cell_here = neigh_new[3]
+
                     new_coords = [
                         site_here.frac_coords[0] + float(cell_here[0]),
                         site_here.frac_coords[1] + float(cell_here[1]),
                         site_here.frac_coords[2] + float(cell_here[2]),
                     ]
                     coords.append(site_here.lattice.get_cartesian_coords(new_coords))
 
-                    # new_site = PeriodicSite(
-                    #     species=site_here.species_string,
-                    #     coords=site_here.lattice.get_cartesian_coords(new_coords),
-                    #     lattice=site_here.lattice,
-                    #     to_unit_cell=False,
-                    #     coords_are_cartesian=True,
-                    # )
                     neighbors_by_distance.append(neigh_new[0])
                     list_distances.append(neigh_new[1])
+                    translations_by_distance.append(cell_here)
                 _list_neighsite = []
                 _list_neighisite = []
                 copied_neighbors_from_ICOHPs = copy.copy(neighbors_from_ICOHPs)
                 copied_distances_from_ICOHPs = copy.copy(lengths_from_ICOHPs)
+                copied_translations_from_ICOHPs = copy.copy(translations_ICOHPs)
                 _neigh_coords = []
                 _neigh_frac_coords = []
 
                 for neigh_idx, neigh in enumerate(neighbors_by_distance):
                     index_here2 = index_here_list[neigh_idx]
-
+                    print(index_here2)
                     for dist_idx, dist in enumerate(copied_distances_from_ICOHPs):
                         if (
                             np.isclose(dist, list_distances[neigh_idx], rtol=1e-4)
                             and copied_neighbors_from_ICOHPs[dist_idx] == index_here2
+                            and (
+                                (
+                                    copied_translations_from_ICOHPs[dist_idx][0]
+                                    == -translations_by_distance[neigh_idx][0]
+                                    and copied_translations_from_ICOHPs[dist_idx][1]
+                                    == -translations_by_distance[neigh_idx][1]
+                                    and copied_translations_from_ICOHPs[dist_idx][2]
+                                    == -translations_by_distance[neigh_idx][2]
+                                )
+                                or (
+                                    copied_translations_from_ICOHPs[dist_idx][0]
+                                    == translations_by_distance[neigh_idx][0]
+                                    and copied_translations_from_ICOHPs[dist_idx][1]
+                                    == translations_by_distance[neigh_idx][1]
+                                    and copied_translations_from_ICOHPs[dist_idx][2]
+                                    == translations_by_distance[neigh_idx][2]
+                                )
```

Comment:
We could raise a warning if this fall-back option is taken. I think with updating to reading the CONTCAR, most issues should be fixed, however.